metricbeat/module/kibana/stats/data_xpack.go

```diff
@@ -0,0 +1,195 @@
+			"snapshot":          c.Bool("snapshot"),
+			"status":            c.Str("status"),
+		}),
+		"usage": c.Dict("usage", s.Schema{
+			"index": c.Str("kibana.index"),
+			"index_pattern": c.Dict("kibana.index_pattern", s.Schema{
+				"total": c.Int("total"),
+			}),
+			"search": c.Dict("kibana.search", s.Schema{
+				"total": c.Int("total"),
+			}),
+			"visualization": c.Dict("kibana.visualization", s.Schema{
+				"total": c.Int("total"),
+			}),
+			"dashboard": c.Dict("kibana.dashboard", s.Schema{
+				"total": c.Int("total"),
+			}),
+			"timelion_sheet": c.Dict("kibana.timelion_sheet", s.Schema{
+				"total": c.Int("total"),
+			}),
+			"graph_workspace": c.Dict("kibana.graph_workspace", s.Schema{
+				"total": c.Int("total"),
+			}),
+			"xpack": s.Object{
+				"reporting": c.Dict("reporting", s.Schema{
+					"available":     c.Bool("available"),
+					"enabled":       c.Bool("enabled"),
+					"browser_type":  c.Str("browser_type"),
+					"_all":          c.Int("all"),
+					"csv":           reportingCsvDict,
+					"printable_pdf": reportingPrintablePdfDict,
+					"status":        reportingStatusDict,
+					"lastDay":       c.Dict("last_day", reportingPeriodSchema, c.DictOptional),
+					"last7Days":     c.Dict("last_7_days", reportingPeriodSchema, c.DictOptional),
+				}, c.DictOptional),
+			},
+		}),
+	}
+
+	reportingCsvDict = c.Dict("csv", s.Schema{
+		"available": c.Bool("available"),
+		"total":     c.Int("total"),
+	}, c.DictOptional)
+
+	reportingPrintablePdfDict = c.Dict("printable_pdf", s.Schema{
+		"available": c.Bool("available"),
+		"total":     c.Int("total"),
+		"app": c.Dict("app", s.Schema{
+			"visualization": c.Int("visualization"),
+			"dashboard":     c.Int("dashboard"),
+		}, c.DictOptional),
+		"layout": c.Dict("layout", s.Schema{
+			"print":           c.Int("print"),
+			"preserve_layout": c.Int("preserve_layout"),
+		}, c.DictOptional),
+	}, c.DictOptional)
+
+	reportingStatusDict = c.Dict("status", s.Schema{
+		"completed":  c.Int("completed", s.Optional),
+		"failed":     c.Int("failed", s.Optional),
+		"processing": c.Int("processing", s.Optional),
+		"pending":    c.Int("pending", s.Optional),
+	}, c.DictOptional)
+
+	reportingPeriodSchema = s.Schema{
+		"_all":          c.Int("all"),
+		"csv":           reportingCsvDict,
+		"printable_pdf": reportingPrintablePdfDict,
+		"status":        reportingStatusDict,
+	}
+)
+
+func reportErrorForMissingField(field string, r mb.ReporterV2) error {
+	err := fmt.Errorf("Could not find field '%v' in Kibana stats API response", field)
+	r.Error(err)
+	return err
+}
+
+func eventMappingXPack(r mb.ReporterV2, intervalMs int64, content []byte) error {
+	var data map[string]interface{}
+	err := json.Unmarshal(content, &data)
+	if err != nil {
+		r.Error(err)
+		return err
+	}
+
+	kibanaStatsFields, err := schemaXPackMonitoring.Apply(data)
+	if err != nil {
+		r.Error(err)
+		return err
+	}
+
+	process, ok := data["process"].(map[string]interface{})
```

Comment:
@jsoriano I think this is an interesting example where we could potentially use "required" fields and skip this part if some required fields are missing. Not part of this PR.